docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -178,38 +190,39 @@ return (
-
-It will disable all the features related to the row grouping, even if a model is provided.
+To disable row grouping for all columns, set the `disableRowGrouping` prop to `true`.
+This disables all features related to row grouping, even if a model is provided.
 
 {{"demo": "RowGroupingDisabled.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### For some columns
+### For specific columns
+
+To disable grouping for a specific column, set the `groupable` property on its `GridColDef` to `false`.
 
-In case you need to disable grouping on specific column(s), set the `groupable` property on the respective column definition (`GridColDef`) to `false`.
-In the example below, the `director` column cannot be grouped. In all examples, the `title` and `gross` columns cannot be grouped.
+In the example below, the **Director** column cannot be grouped, even though there are repeating values.
+(The **Title** and **Gross** columns cannot be grouped in any examples in this doc because there are no repeating values.)
 
 {{"demo": "RowGroupingColDefCanBeGrouped.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Grouping non-groupable columns programmatically
+### Grouping non-groupable columns
 
-To apply row grouping programmatically on non-groupable columns (columns with `groupable: false` in the [column definition](/x/api/data-grid/grid-col-def/)), you can provide row grouping model in one of the following ways:
+To apply row grouping programmatically on non-groupable columns (columns with `groupable: false` in the `GridColDef`), you can provide the row grouping model in one of three ways:
 
-1. Pass `rowGrouping.model` to the `initialState` prop. This will [initialize the grouping](/x/react-data-grid/row-grouping/#initialize-the-row-grouping) with the provided model.
-2. Provide the `rowGroupingModel` prop. This will [control the grouping](/x/react-data-grid/row-grouping/#controlled-row-grouping) with the provided model.
-3. Call the API method `setRowGroupingModel`. This will set the aggregation with the provided model.
+1. Pass `rowGrouping.model` to the `initialState` prop. This [initializes grouping](/x/react-data-grid/row-grouping/#initializing-row-grouping) with the provided model.
+2. Provide the `rowGroupingModel` prop. This [controls grouping](/x/react-data-grid/row-grouping/#controlled-row-grouping) with the provided model.
+3. Call the API method `setRowGroupingModel`. This sets the aggregation with the provided model.
 
-In the following example, the column `company` is not groupable from the UI but the `rowGroupingModel` prop is passed to generate a read-only row group.
+In the following example, the **Company** column is not groupable through the interface, but the `rowGroupingModel` prop is passed to generate a read-only row group.
 
 {{"demo": "RowGroupingReadOnly.js", "bg": "inline", "defaultCodeOpen": false}}
 
-## Using `groupingValueGetter` for complex grouping value
+## Complex grouping values
```

Comment:
Debatable, but one benefit of more detailed headings like the previous one could be to get nice search results on Algolia:

<img width="660" alt="image" src="https://github.com/user-attachments/assets/a9bb5de4-f8a1-4993-917f-a9065f428f3c" />

Reply:
Good call, I'm usually trying to optimize for quickly scanning the table of contents nav when I make changes like this, but search is important to factor in as well. I'll revert this one and reevaluate the other headers here.

And I always add redirects when page slugs change, but it never occurred to me to do it for header links too! 😱

Reply:
Oh, I was wrong about the redirects for anchors (at least server-side redirects) since the hash is not sent to the server.
You can redirect **to** URL with a hash, but you can't redirect **from** URL with a hash, and this is where my confusion came from – sorry about that! 😅